BALSAMIC/assets/scripts/collect_qc_metrics.py

```diff
@@ -0,0 +1,138 @@
+@click.argument("sequencing_type", required=True)
+@click.argument("capture_kit", required=True)
+def collect_qc_metrics(output_path, multiqc_data_path, sequencing_type, capture_kit):
+    """Extracts the requested metrics and saves them to the output path"""
+
+    with open(output_path, "w") as fn:
+        yaml.dump(
+            get_multiqc_metrics(multiqc_data_path, sequencing_type, capture_kit),
+            fn,
+            sort_keys=False,
+            default_flow_style=False,
+        )
+
+
+def get_multiqc_data_source(data, sample, source_name):
+    """Extracts the metrics data source associated with sample and source names"""
+
+    # Splits multiqc_picard_dups into ['multiqc', 'picard', 'dup'] in order to retrieve the
+    # ["report_data_sources"]["Picard"]["DuplicationMetrics"] values from multiqc_data.json
+    source = source_name[:-1].split("_")
+
+    # Nested json fetching
+    for source_tool in data["report_data_sources"]:
+        for source_step in data["report_data_sources"][source_tool]:
+            if (
+                source[1].lower() in source_tool.lower()
+                and source[2].lower() in source_step.lower()
+            ):
+                try:
+                    return os.path.basename(
+                        data["report_data_sources"][source_tool][source_step][sample]
+                    )
+                except KeyError:
+                    # Deletes par orientation information from the sample name (insertSize metrics)
+                    sample = sample.rsplit("_", 1)[0]
+
+                    return os.path.basename(
+                        data["report_data_sources"][source_tool][source_step][sample]
+                    )
+
+
+def get_qc_available_panel_beds(metrics):
+    """Returns available panel bed file names"""
+    available_beds = []
+
+    for k in metrics:
+        if k != "default":
+            available_beds.append(k)
+
+    return available_beds
+
+
+def merge_dicts(*dicts):
+    """Merges multiple dictionaries integrating by common keys"""
+    merged_dict = {}
+
+    for d in dicts:
+        for key in d:
+            try:
+                # Overwrites the default values with panel specific ones
+                merged_dict[key].update(d[key])
+            except KeyError:
+                merged_dict[key] = d[key]
+
+    return merged_dict
+
+
+def get_requested_metrics(metrics, sequencing_type, capture_kit):
+    """Parses the requested metrics and returns them as a dictionary"""
+
+    if sequencing_type != "wgs" and capture_kit in get_qc_available_panel_beds(
+        metrics[sequencing_type]
+    ):
+        requested_metrics = merge_dicts(
+            metrics[sequencing_type]["default"],
+            metrics[sequencing_type][capture_kit],
+        )
+    elif sequencing_type != "wgs":
+        requested_metrics = metrics[sequencing_type]["default"]
+    else:
+        requested_metrics = metrics[sequencing_type]
```

Comment:
```suggestion
    # Assume it is wgs
    requested_metrics = metrics[sequencing_type]
    if capture_kit:
        requested_metrics = metrics[sequencing_type]["default"]
        if capture_kit in get_qc_available_panel_beds(metrics[sequencing_type]):
            requested_metrics = merge_dicts(
                metrics[sequencing_type]["default"],
                metrics[sequencing_type][capture_kit],
             )
```

Second if can also be moved out. `wgs` and `capturekit` never overlap. So no need to check it.

Reply:
Updated 👍